Initialize springboot with gradle

chin8628 opened the pull request:
Initialize Springboot with Gradle.

chin8628 committed:
Initialize springboot with gradle

champillon commented on src/test/java/io/group3/backend/EtaxBackendApplicationTests.java:
```diff
@@ -0,0 +1,16 @@
+public class EtaxBackendApplicationTests {
+
+	@Test
+	public void contextLoads() {
```
ทำไมมี test ปล่าวหล่ะครับ??? ถ้าจะยังไม่ทำอะไรให้ใส่ // TODO ไว้ครับ แต่พี่ว่าควรมีตัวอย่าง test ในการ Initial Project นะครับ

chin8628 replied:
ตรงนี้เดี๋ยวผมขอใส่ // TODO ไว้ก่อนละกันนะครับ เพราะยังไม่ได้ตรงลงกันภายในทีมว่าจะมีการ Test กันแบบไหน

champillon replied:
ได้ครับ

champillon commented on build.gradle:
```diff
@@ -0,0 +1,29 @@
+apply plugin: 'org.springframework.boot'
+
+group = 'io.group3'
+version = '0.0.1-SNAPSHOT'
```
อยากทราบวิธี run version ของกลุ่มนี้ครับ และน่าจะเพิ่มใน CONVENTION.md ด้วยครับ

champillon commented on src/main/java/io/group3/backend/controller/Hello.java:
```diff
@@ -0,0 +1,21 @@
+import java.util.concurrent.atomic.AtomicInteger;
+
+@RestController
+public class Hello {
```
อันนี้ไว้ทำอะไรครับ? ถ้าเป็น Controller ควรตั้งชื่อเป็น HelloController จะดีกว่าครับ จะได้ไม่สับสนกับ layer อื่นของ Application

chin8628 replied:
สงสัยเคสนี้ครับว่า ในเมื่อไฟล์อยู่ภายในแพกเกจของ controller อยู่แล้วเราต้องระบุในชื่อไฟล์ซ้ำอีกรอบหรอครับ

champillon replied:
งั้นเอาแบบนี้ไปก่อนก็ได้ครับ ไว้เดี๋ยวเรียนเรื่อง design pattern แล้วค่อยว่ากันอีกที

mahasak commented:
@chin8628 Why we put binary in to repository ?
please exclude it in .gitignore

mahasak commented on src/main/java/io/group3/backend/model/Greeting.java:
```diff
@@ -0,0 +1,21 @@
+        return id;
+    }
+
+}
```
I'm not fan of getter/settter. there're alternative way to us it as POJO. Do we interest in changing to that ?

chin8628 replied:
Thank you for the suggestion. I've already updated our model file by remove getter/setter and changed access modified of properties to Public.

XIIKJIIX commented:
@mahasak Do you refer to`gradle-wrapper.jar`? It needs to be used by `gradlew` and `gradlew.bat` build script. For example, [this line](https://github.com/it-kmitl-2018/group3-backend/blob/95ec67a7431478976dcb628c33e9b9eaa21990e6/gradlew#L67). 
Besides, [SPRING INITIALIZR](https://start.spring.io/) is generate `.gitignore` for us.

champillon commented on build.gradle:
```diff
@@ -0,0 +1,29 @@
+apply plugin: 'eclipse'
+apply plugin: 'org.springframework.boot'
+
+group = 'io.group3'
```
อันนี้น่าจะเป็น th.ac.kmitl.it.soa.group3 รึป่าวครับ

champillon reviewed (changes requested):
ฝากแก้ด้วยนะครับ

chin8628 committed:
Re-initialize spring boot with grandle

- Remove getter/setter from model
- remove placeholder code in test files
- update version number, accroding to convention team.
- rename root package

chin8628 committed:
Rename Greeting file to GreetingModel.

champillon commented on src/test/java/th/ac/kmitl/it/soa/group3/EtaxBackendApplicationTests.java:
```diff
@@ -0,0 +1,14 @@
+@SpringBootTest
+public class EtaxBackendApplicationTests {
+
+	// TODO
```
น้องช่วยเพิ่มด้วยครับ ว่า TODO ที่น้องต้องการที่ กำลังจะทำอะไร (TODO => to do หรือสิ่งที่จะทำครับ)

chin8628 replied:
คอมมิตใหม่แล้วครับ ขอโทษครับพอดีเบลอเลยลืมเขียน

champillon reviewed (commented):
ฝากชี้แจงด้วยครับ

chin8628 committed:
Update TODO in test file.

champillon commented:
@XIIKJIIX ลองลบ gradle* ออก ยกเว้น gradle.build ครับ ที่เหลือ gradle.build จะไปโหลดมาให้เอง

XIIKJIIX committed:
Remove gradle folder

XIIKJIIX committed:
Add gradle folder to .gitignore

XIIKJIIX commented:
@champillon ลบออกแล้วนะครับ

champillon commented:
@XIIKJIIX gradlew กับ gradlew.bat ด้วยจร๊ะ เดี๋ยว build.gradle จะไปโหลดมาให้

XIIKJIIX committed:
Put gradle* back to version control

chin8628 commented:
@mahasak Could you review this pr again?

champillon commented:
merged ให้เลย ถ้าอันไหนค้างนานทวงได้ ถือว่า 1 approve Ok แล้ว ถ้าเกิน 3 วัน

champillon commented:
คะแนนครับ 2 คะแนน
5.1 การ design เป็น components เพื่อทำให้คนอื่นทำงานร่วมกันต่อได้ง่าย => 1 คะแนน มีการแบ่ง Model กับ Controller
5.2 คุณภาพของการตั้งชื่อตัวแปร, ฟังก์ชั่น และความเรียบร้อยของ code => 0 คะแนน GreetingModel กับ HelloController ยังไม่สื่อ
5.3 ความยากเชิงเทคนิคที่นำมาประยุกต์ใช้ เช่น แปลง XML Schema เป็น Java code => 0 คะแนน ยังไม่มีอะไรยาก
5.4 รายละเอียดการตั้งชื่อใน PR name และ commit log รวมทั้งรายละเอียดที่บรรยายใน issue ของ PR นั้นๆ => 1 เขียน PR name และ commit log ได้ดี
5.5 ความสอดคล้องกันกับ PR อื่นๆของทีม ที่ทำงานร่วมกันได้ดี => 0 ยังไม่เห็นการทำงานร่วมกันเพื่อนๆ
